sklearn/datasets/kddcup99.py

```diff
@@ -0,0 +1,342 @@
+        Each row corresponds to the 54XXX features in the dataset.
+
+    dataset.target : numpy array of shape (581012,) XXX
+        Each value corresponds to one of the 21 attack types or to the label
+        'normal.'.
+
+    dataset.DESCR : string
+        Description of the kddcup99 dataset.
+
+    """
+
+    data_home = get_data_home(data_home=data_home)
+    if sys.version_info[0] == 3:
+        # The zlib compression format use by joblib is not compatible when
+        # switching from Python 2 to Python 3, let us use a separate folder
+        # under Python 3:
+        dir_suffix = "-py3"
+    else:
+        # Backward compat for Python 2 users
+        dir_suffix = ""
+    kddcup_dir = join(data_home, "kddcup99" + dir_suffix)
+    samples_path = join(kddcup_dir, "samples")
+    targets_path = join(kddcup_dir, "targets")
+    available = exists(samples_path)
+
+    if download_if_missing and not available:
+        _mkdirp(kddcup_dir)
+        logger.warning("Downloading %s" % URL)
+        f = BytesIO(urlopen(URL).read())
+        Xy = pd.io.parsers.read_csv(GzipFile(fileobj=f), engine='c',
+                                    header=None)
+        Xy = np.asarray(Xy)
+
+        X = Xy[:, :-1]
+        y = Xy[:, -1]
+
+        joblib.dump(X, samples_path, compress=9)
+        joblib.dump(y, targets_path, compress=9)
+
+    try:
+        print kddcup_dir
+        X, y
+    except NameError:
+        X = joblib.load(samples_path)
+        y = joblib.load(targets_path)
+
+    if shuffle:
+        ind = np.arange(X.shape[0])
+        rng = check_random_state(random_state)
+        rng.shuffle(ind)
+        X = X[ind]
+        y = y[ind]
+
+    return Bunch(data=X, target=y, DESCR=__doc__)
+
+
+def _mkdirp(d):
+    """Ensure directory d exists (like mkdir -p on Unix)
+    No guarantee that the directory is writable.
+    """
+    try:
+        os.makedirs(d)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
+
+
+# XXX: to be moved (pb if moved in .base: circular importation with mldata)
+def load_shuttle(shuffle=False, random_state=None):
+    """Load and return the shuttle dataset (classification - anomaly detection).
+
+    The Shuttle dataset is the fusion of the training and testing datasets
+    available in the UCI repository.
+    The shuttle dataset contains 9 attributes all of which are numerical.
+    the first one being time.  The last column is the class which has been
+    coded as follows :
+        1       Rad Flow
+        2       Fpv Close
+        3       Fpv Open
+        4       High
+        5       Bypass
+        6       Bpv Close
+        7       Bpv Open
+
+    Approximately 80% of the data belongs to class 1.
+
+    =================   ==============
+    Classes                          7
+    Samples total                  58000
+    Dimensionality                   9
+    Targets            int, 1,2,...,7
+    =================   ==============
```

Comment:
see Simple table: in
http://docutils.sourceforge.net/docs/user/rst/quickref.html#tables

you should respect columns widths and alignment of characters